sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -163,13 +163,75 @@ internal string OrderByRaw
         public IList<string> ScoringParameters { get; internal set; } = new List<string>();
 
         /// <summary> A value that specifies the language of the search query. </summary>
-        internal QueryLanguage? QueryLanguage { get; set; }
+        [CodeGenMember("queryLanguage")]
+        public QueryLanguage? QueryLanguage { get; set; }
 
-        /// <summary> A value that specified the type of the speller to use to spell-correct individual search query terms. </summary>
-        internal Speller? Speller { get; set; }
+        /// <summary> A value that specifies the type of the speller to use to spell-correct individual search query terms. </summary>
+        [CodeGenMember("speller")]
+        public QuerySpeller? QuerySpeller { get; set; }
 
-        /// <summary> A value that specifies whether answers should be returned as part of the search response. </summary>
-        internal Answers? Answers { get; set; }
+        /// <summary> A value that specifies whether <see cref="SearchResults{T}.Answers"/> should be returned as part of the search response. </summary>
+        public QueryAnswer? QueryAnswer { get; set; }
+
+        /// <summary>
+        /// A value that specifies the number of <see cref="SearchResults{T}.Answers"/> that should be returned as part of the search response.
+        /// <para>The default is 1 and the maximum is 5.</para>
+        /// </summary>
+        public uint QueryAnswerCount { get; set; }
+
+        /// <summary>
+        /// Join OrderBy so it can be sent as a comma separated string.
+        /// </summary>
+        [CodeGenMember("answers")]
+        internal string QueryAnswerRaw
+        {
+            get
+            {
+                if (!QueryAnswer.HasValue)
+                {
+                    return null;
+                }
+
+                string queryAnswerStringValue = QueryAnswer.Value.ToString();
+                if (QueryAnswer.Value == Models.QueryAnswer.Extractive)
+                {
+                    queryAnswerStringValue = $"{Models.QueryAnswer.Extractive}|count-{Math.Min(5, QueryAnswerCount)}";
```

Comment:
I worry this is too magical, but let's leave it for now and think about it more before GA.